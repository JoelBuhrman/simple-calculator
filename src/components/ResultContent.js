import React from 'react'
import { SUM, MULTIPLY } from '../constants'

const ResultContent = ({ method, result, updateMethod }) => (
  <div>
    <input
      type='radio'
      checked={method === SUM}
      onClick={() => updateMethod(SUM)}
    />
    Sum
    <br />
    <input
      type='radio'
      checked={method === MULTIPLY}
      onClick={() => updateMethod(MULTIPLY)}
    />
    Multiply
    <br />
    <br />
    {`Result: ${result}`}
  </div>
)

export default ResultContent
