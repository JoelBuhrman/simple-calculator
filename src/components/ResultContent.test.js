import React from 'react'
import TestRenderer from 'react-test-renderer'
import ResultContent from './ResultContent'

it('renders without crashing', () => {
  const snapshot = TestRenderer.create(
    <ResultContent method={'MULTIPLY'} result={0} updateMethod={jest.fn()} />
  ).toJSON()
  expect(snapshot).toMatchSnapshot()
})
