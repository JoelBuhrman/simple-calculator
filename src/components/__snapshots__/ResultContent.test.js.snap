// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <input
    checked={false}
    onClick={[Function]}
    type="radio"
  />
  Sum
  <br />
  <input
    checked={false}
    onClick={[Function]}
    type="radio"
  />
  Multiply
  <br />
  <br />
  Result: 0
</div>
`;
